FrontEnd/src/UserAuthentication/SignUpPage.jsx

```diff
@@ -1,77 +1,124 @@
-import  {useNavigate} from "react-router-dom";
-import {useState} from "react";
-import {registerUser} from "./Auth";
-import {addUser} from './FirestoreDB';
-import './SignUpPage.css';
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { registerUser } from "./Auth";
+import "./SignUpPage.css";
 
-function SignUpPage(){
-    const navigate = useNavigate();
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
-    const [firstName, setFirstName] = useState('');
-    const [lastName, setLastName] = useState('');
-    const [userName, setUserName] = useState('');
-    const MIN_PASSWORD_LENGTH = 6;
-    const handleLoginRoute = () => {
-        navigate(`/LoginPage`);
-      };
+function SignUpPage() {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [userName, setUserName] = useState("");
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const MIN_PASSWORD_LENGTH = 6;
+  const handleLoginRoute = () => {
+    navigate(`/LoginPage`);
+  };
 
-    const handleSignUp = async(event) =>{
-        event.preventDefault();
-        if (password.length < MIN_PASSWORD_LENGTH) {
-            alert(`Minimum password length is ${MIN_PASSWORD_LENGTH}`);
-        } else {
-            try{
-                const userCredential = await registerUser(email, password);
-                if (userCredential!==undefined){
-                    addUser(userCredential, firstName, lastName, userName, email);
-                    navigate('/MainPage');
-                } else{
-                    alert("Incorrect Email address");
-                };
-            } catch (error){
-                console.error("Error registering user:", error);
-            };
-        };
-    };
-    return(
+  const handleSignUp = async (event) => {
+    event.preventDefault();
+    if (password.length < MIN_PASSWORD_LENGTH) {
+      alert(`Minimum password length is ${MIN_PASSWORD_LENGTH}`);
+    } else {
+      try {
+        const userCredential = await registerUser(
+          email,
+          password,
+          userName,
+          firstName,
+          lastName
+        );
+        if (userCredential !== null) {
+          alert("Account successfully created");
```

Comment:
Probably don't want to use `alert` here, it's gonna block the code execution. You can redirect a user to their profile page, so it's clear the signup was successful.